libnavigation-util/src/test/java/com/mapbox/navigation/utils/thread/ThreadControllerKtTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.utils.thread
+
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ClosedReceiveChannelException
+import kotlinx.coroutines.runBlocking
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class ThreadControllerKtTest {
+
+    @Test
+    fun monitorChannelWithException_callsOnCancellation_whenChannelClosed() {
+        var flag = false
+        val channel: Channel<String> = Channel()
+
+        runBlocking {
+            monitorChannelWithException(
+                    channel,
+                    {},
+                    { flag = true }
+            )
+
+            channel.send("foobar")
+
+            assertFalse(flag)
+
+            channel.close(ClosedReceiveChannelException(""))
+            try {
+                channel.send("foobar")
+            } catch (ex: Exception) {}
+        }
+        assertTrue(flag)
+    }
+
+    @Test
+    fun monitorChannelWithException() {
+        val channel: Channel<String> = Channel()
+        var msg = "error"
+
+        runBlocking {
+            monitorChannelWithException(
+                    channel,
+                    {
+                        msg = it
+                    },
+                    {}
+            )
+
+            channel.send("success")
+            channel.close()
+        }
+
+        assertThat(msg, `is`("success"))
```

Comment:
Do we need Hamcrest here? IMO it doesn't add too much value in this simple case.

```suggestion
        assertEquals("success", msg)
```

What do you think?

Reply:
This was more out of habit than anything else. I'll change it.  May I suggest we bring in the Kluent library?  So we can write assertions like: "hello" shouldBeEqualTo "hello"

https://github.com/MarkusAmshove/Kluent